mlflow/transformers/__init__.py

```diff
@@ -1112,21 +1122,62 @@ def _fetch_model_card(model_or_pipeline):
         )
         return
 
-    model = model_or_pipeline.model
-
     if hasattr(hub, "ModelCard"):
         try:
-            return hub.ModelCard.load(model.name_or_path)
+            return hub.ModelCard.load(model_name)
         except Exception as e:
             _logger.warning(f"The model card could not be retrieved from the hub due to {e}")
     else:
         _logger.warning(
-            f"The version of huggingface_hub that is installed does not provide "
+            "The version of huggingface_hub that is installed does not provide "
             f"ModelCard functionality. You have version {hub.__version__} installed. "
-            f"Update huggingface_hub to >= '0.10.0' to retrieve the ModelCard data."
+            "Update huggingface_hub to >= '0.10.0' to retrieve the ModelCard data."
         )
 
 
+def _fetch_license(model_name, card_data):
+    """
+    Attempts to retrieve a license document from the underlying model repository iff the
+    `huggingface_hub` library is installed. If the library is available and a license file
+    is present, return the contents as text. If the license file does not exist, return
+    instructions for a user to investigate what the underlying usage restrictions that
+    exist.
+    """
+
+    fallback = (
+        f"A license file could not be found for the '{model_name}' repository. \n"
+        "To ensure that you are in compliance with the license requirements for this "
+        f"model, please visit the model repository here:\n https://huggingface.co/{model_name}"
+    )
+
+    def _get_license_file_as_text(repo_id):
+        for license_name in _LICENSE_FILE_NAMES:
+            try:
+                license_file_local_location = hub.hf_hub_download(
+                    repo_id=repo_id, filename=license_name
+                )
+                return pathlib.Path(license_file_local_location).read_text()
```

Comment:
shall we delete the temporary `license_file_local_location` file before returning ?

Reply:
the huggingface_hub SDK will put the license file in with the rest of the automated GC-like behavior for the file system location (it's a disk cache). Technically, our logic here is actually fetching from the cache first, but I don't want to write code for finding that location directly as it's fairly complex and involves low-level APIs that aren't public). When the local disk cache gets too full, the hub SDK, before downloading a new model, will cull the repo directory least frequently accessed along with all of the model files.